cpp/tests/copying/shift_tests.cu

```diff
@@ -0,0 +1,175 @@
+
+TYPED_TEST(ShiftTest, MisMatchFillValueCount)
+{
+    using T = TypeParam;
+
+    fixed_width_column_wrapper<T> a{};
+
+    cudf::table_view input{ { a } };
+
+    std::vector<std::unique_ptr<cudf::scalar>> fills{};
+    fills.push_back(make_scalar<T>());
+
+    cudf::experimental::shift(input, 5, fills);
+}
+
+TYPED_TEST(ShiftTest, TwoColumnsEmpty)
+{
+    using T =  TypeParam;
+
+    fixed_width_column_wrapper<T> a{};
+    fixed_width_column_wrapper<T> b{};
+
+    cudf::table_view input{ { a, b } };
+    std::vector<std::unique_ptr<cudf::scalar>> fills{};
+    fills.push_back(make_scalar<T>());
+    fills.push_back(make_scalar<T>());
+    cudf::experimental::shift(input, 5, fills);
+}
+
+TYPED_TEST(ShiftTest, OneColumn)
+{
+    using T =  TypeParam;
+
+    auto input_a = fixed_width_column_wrapper<T>{ lowest<T>, 1, 2, 3, 4, 5, highest<T> };
+    auto input = cudf::table_view { { input_a } };
+
+    auto expected_a = fixed_width_column_wrapper<T>{ 7, 7, lowest<T>, 1, 2, 3, 4 };
+    auto expected = cudf::table_view{ { expected_a } };
+
+    std::vector<std::unique_ptr<cudf::scalar>> fills {};
+    fills.push_back(make_scalar<T>(7));
+
+    auto actual = cudf::experimental::shift(input, 2, fills);
+
+    cudf::test::expect_tables_equal(expected, *actual);
+}
+
+TYPED_TEST(ShiftTest, OneColumnNegativeShift)
+{
+    using T =  TypeParam;
+
+    auto input_a = fixed_width_column_wrapper<T>{ lowest<T>, 1, 2, 3, 4, 5, highest<T> };
+    auto input = cudf::table_view { { input_a } };
+
+    auto expected_a = fixed_width_column_wrapper<T>{ 4, 5, highest<T>, 7, 7, 7, 7 };
+    auto expected = cudf::table_view{ { expected_a } };
+
+    std::vector<std::unique_ptr<cudf::scalar>> fills {};
+    fills.push_back(make_scalar<T>(7));
+
+    auto actual = cudf::experimental::shift(input, -4, fills);
+
+    cudf::test::expect_tables_equal(expected, *actual);
+}
+
+TYPED_TEST(ShiftTest, OneColumnNullFill)
+{
+    using T =  TypeParam;
+
+    auto input_a = fixed_width_column_wrapper<T>{ lowest<T>, 5, 0, 3, 0, 1, highest<T> };
+    auto input = cudf::table_view { { input_a } };
+
+    auto expected_a = fixed_width_column_wrapper<T>({ 0, 0, lowest<T>, 5, 0, 3, 0 }, { 0, 0, 1, 1, 1, 1, 1 });
+    auto expected = cudf::table_view{ { expected_a } };
+
+    std::vector<std::unique_ptr<cudf::scalar>> fills {};
+    fills.push_back(make_scalar<T>());
+
+    auto actual = cudf::experimental::shift(input, 2, fills);
+
+    cudf::test::expect_tables_equal(expected, *actual);
+}
+
+TYPED_TEST(ShiftTest, TwoColumnsNullableInput)
+{
+    using T =  TypeParam;
+
+    auto input_a = fixed_width_column_wrapper<T>({ 1, 2, 3, 4, 5 }, { 0, 1, 1, 1, 0});
+    auto input_b = fixed_width_column_wrapper<T>({ 5, 4, 3, 2, 1 }, { 1, 0, 1, 1, 1});
+    auto input = cudf::table_view { { input_a, input_b } };
+
+    auto expected_a = fixed_width_column_wrapper<T>({ 7, 7, 1, 2, 3 }, { 1, 1, 0, 1, 1});
+    auto expected_b = fixed_width_column_wrapper<T>({ 7, 7, 5, 4, 3 }, { 0, 0, 1, 0, 1});
```

Comment:
Wouldn't this be `{ 5, 4, 5, 4, 3 }` since scalar b is null? Guess it doesn't matter since null values aren't checked for equality